cellxgene_resources/compare_revision.ipynb

```diff
@@ -80,118 +36,92 @@
+    "        if k not in should_be_absent:\n",
+    "            print('not present: ' + k)\n",
+    "    elif collection.get(k) != v and k not in should_differ_collection:\n",
+    "        if k == 'datasets':\n",
+    "            diff_props = []\n",
+    "            pub_datasets = {d['dataset_id']: d for d in collection[k]}\n",
+    "            rev_datasets = {d['dataset_id']: d for d in rev[k]}\n",
+    "            comp = {}\n",
+    "            for ds_id,v in rev_datasets.items():\n",
+    "                comp[ds_id] = {'title': v['title']}\n",
+    "                if ds_id not in pub_datasets.keys():\n",
+    "                    comp[ds_id]['in published'] = 'not present'\n",
+    "                else:\n",
+    "                    for prop,rev_val in v.items():\n",
+    "                        if prop not in should_differ_dataset:\n",
+    "                            pub_val = pub_datasets[ds_id].get(prop)\n",
+    "                            if prop in ont_fields:\n",
+    "                                rev_val = [t['label'] for t in rev_val]\n",
+    "                                pub_val = [t['label'] for t in pub_val]\n",
+    "                            if isinstance(rev_val, list) and prop != 'assets':\n",
+    "                                rev_val.sort()\n",
+    "                                pub_val.sort()\n",
+    "                            if pub_val != rev_val:\n",
+    "                                if prop == 'mean_genes_per_cell' and round(rev_val, 5) == round(pub_val, 5):\n",
+    "                                    continue\n",
+    "                                diff_props.append(prop)\n",
+    "                                comp[ds_id][prop + '_REV'] = rev_val\n",
+    "                                comp[ds_id][prop + '_PUB'] = pub_val\n",
+    "        else:\n",
+    "            print('not same: ' + k)\n",
+    "            if k not in ['datasets','publisher_metadata']:\n",
+    "                if k in ['links']:\n",
+    "                    diff_in_pub = [l for l in collection[k] if l not in v]\n",
+    "                    print('--- published: ', diff_in_pub)\n",
+    "                    diff_in_rev = [l for l in v if l not in collection[k]]\n",
+    "                    print('----- revised: ', diff_in_rev)\n",
+    "                else:\n",
+    "                    print('--- published: ', str(collection[k]))\n",
+    "                    print('----- revised: ', v)\n",
+    "\n",
+    "\n",
+    "comp_df = pd.DataFrame(comp).transpose()\n",
+    "comp_df = comp_df.dropna(subset=[c for c in comp_df.columns if c != 'title'], how='all')\n",
+    "if not comp_df.empty:\n",
+    "    display(comp_df.fillna(''))\n",
+    "\n",
+    "for f in diff_props:\n",
     "    temp = comp_df[(comp_df[f + '_REV'] != comp_df[f + '_PUB']) & (comp_df[f + '_PUB'].isna() == False)]\n",
     "    for i,row in temp.iterrows():\n",
     "        print(i + '-' + f)\n",
```

Comment:
I can't comment on the actual line, but the line with `only_in_pub = [e for e in p if e no in r]` raises a `TypeError: 'int' object is not iterable.` 

Adding something like :
`if isinstance(p, (int, float)): continue`,
will skip over the non-list values and then print out just the differences between the category columns

Reply:
fixed via e7323cbc3a73ce66a41960438639016ebdaced2f